[CI] FieldCapsIT testAllIndicesWithIndexFilter {upgradedNodes=2} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/nbzfldd4k6kb4/tests/:qa:rolling-upgrade:v8.10.4%23bwcTest/org.elasticsearch.upgrades.FieldCapsIT/testAllIndicesWithIndexFilter%20%7BupgradedNodes=2%7D


**Reproduction line:**
```
./gradlew ':qa:rolling-upgrade:v8.10.4#bwcTest' -Dtests.class="org.elasticsearch.upgrades.FieldCapsIT" -Dtests.method="testAllIndicesWithIndexFilter {upgradedNodes=2}" -Dtests.seed=CA481EC5C9EE6022 -Dtests.bwc=true -Dtests.locale=no -Dtests.timezone=Pacific/Honolulu -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.upgrades.FieldCapsIT#testAllIndicesWithIndexFilter {upgradedNodes=2}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testAllIndicesWithIndexFilter%20%7BupgradedNodes%3D2%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.upgrades.FieldCapsIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: ["new_green_1", "new_green_2", "new_red_1", "new_red_2", "old_green_1", "old_green_2", "old_red_1", "old_red_2"]
     but: was ["new_green_2", "new_red_1", "old_green_2", "old_red_2"]

  at __randomizedtesting.SeedInfo.seed([CA481EC5C9EE6022:D85788E533AB272C]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.upgrades.FieldCapsIT.testAllIndicesWithIndexFilter(FieldCapsIT.java:247)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
  at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

Re-labeling because this seems to be a bug with the index template BwC. From the logs we see this AssertionError:
```
[2023-12-14T03:25:14,790][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-2] fatal error in thread [elasticsearch[test-cluster-2][cluster_coordination][T#1]], exiting java.lang.AssertionError: org.elasticsearch.gateway.CorruptStateException: org.elasticsearch.xcontent.XContentParseException: [-1:40410] [index_template] failed to parse field [index_template]
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.assertOnCommit(PersistedClusterStateService.java:1244)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.commit(PersistedClusterStateService.java:1234)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.writeIncrementalStateAndCommit(PersistedClusterStateService.java:976)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.GatewayMetaState$LucenePersistedState.writeClusterStateToDisk(GatewayMetaState.java:579)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.GatewayMetaState$LucenePersistedState.setLastAcceptedState(GatewayMetaState.java:562)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.CoordinationState.handlePublishRequest(CoordinationState.java:392)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.Coordinator.handlePublishRequest(Coordinator.java:469)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.acceptState(PublicationTransportHandler.java:213)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.handleIncomingPublishRequest(PublicationTransportHandler.java:200)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.coordination.PublicationTransportHandler.lambda$new$0(PublicationTransportHandler.java:112)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:74)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:294)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.elasticsearch.gateway.CorruptStateException: org.elasticsearch.xcontent.XContentParseException: [-1:40410] [index_template] failed to parse field [index_template]
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:674)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$6(PersistedClusterStateService.java:572)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:716)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:571)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService$Writer.assertOnCommit(PersistedClusterStateService.java:1242)
	... 16 more
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:40410] [index_template] failed to parse field [index_template]
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplateMetadata.fromXContent(ComposableIndexTemplateMetadata.java:67)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:54)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:147)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:414)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.Metadata$Builder.fromXContent(Metadata.java:2660)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.gateway.PersistedClusterStateService.readXContent(PersistedClusterStateService.java:672)
	... 20 more
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:40410] [index_template] failed to parse field [template]
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.throwFailedToParse(ObjectParser.java:616)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:611)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseSub(ObjectParser.java:629)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:315)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplate.parse(ComposableIndexTemplate.java:102)
	at org.elasticsearch.server@8.10.4/org.elasticsearch.cluster.metadata.ComposableIndexTemplateMetadata.lambda$static$1(ComposableIndexTemplateMetadata.java:50)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
	... 29 more
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:40408] [template] unknown field [lifecycle]
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$errorOnUnknown$2(ObjectParser.java:105)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:299)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:166)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.lambda$declareField$10(ObjectParser.java:431)
	at org.elasticsearch.xcontent@8.10.4/org.elasticsearch.xcontent.ObjectParser.parseValue(ObjectParser.java:609)
	... 36 more
```

--- a/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FieldCapsIT.java
+++ b/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/FieldCapsIT.java
@@ -12,7 +12,6 @@ package org.elasticsearch.upgrades;
 import com.carrotsearch.randomizedtesting.annotations.Name;
 
 import org.apache.http.HttpHost;
-import org.apache.lucene.tests.util.LuceneTestCase.AwaitsFix;
 import org.elasticsearch.Build;
 import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
 import org.elasticsearch.client.Request;
@@ -40,7 +39,6 @@ import static org.hamcrest.Matchers.equalTo;
  * In 8.2 we also added the ability to filter fields by type and metadata, with some post-hoc filtering applied on
  * the co-ordinating node if older nodes were included in the system
  */
-@AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/103473")
 public class FieldCapsIT extends AbstractRollingUpgradeTestCase {
 
     public FieldCapsIT(@Name("upgradedNodes") int upgradedNodes) {

